### ISSUE TITLE
Demo on pre-trained checkpoints for Tiny Imagenet

### ISSUE THREAD

**Issue by nkotelevskii** [ENSTA-U2IS-AI/torch-uncertainty#112]
Hello!

First of all, thank you so much for creating this amazing library and for sharing the pre-trained checkpoints from your recent paper [here](https://arxiv.org/abs/2310.08287). It’s awesome!

I’m currently exploring the pre-trained models for Tiny-Imagenet ([here](https://huggingface.co/datasets/torch-uncertainty/Checkpoints)), but I’ve run into some difficulties while trying to load them. Could you possibly provide a simple demo or example on how to properly load these models and evaluate them on the validation dataset (for example compute accuracy)?

I really appreciate your help and look forward to your guidance!

Thanks again!

**Comment by o-laurent** [comment]
Hi @nkotelevskii, 

Sorry for the delay, and thanks for your kind message. After downloading and untaring the weights, the following function should work:

```python
from pathlib import Path

from torch_uncertainty.models.resnet import resnet
from safetensors.torch import load_file


def load_model(version: int):
    """Load the model corresponding to the given version."""
    model = resnet(
        arch=18,
        num_classes=200,
        in_channels=3,
        style="cifar",
        conv_bias=False,
    )
    path = Path(
        f"tiny-imagenet-resnet18/tiny-imagenet-resnet18-0-1023/version_{version}.safetensors"
    )
    if not path.exists():
        raise ValueError("File does not exist")

    state_dict = load_file(path)
    model.load_state_dict(state_dict=state_dict)
    return model
```
You may have to change the path slightly. Let me know if still struggle with loading the models.

Then, test the accuracy with the following script (you may also wrap the model with torch_uncertainty.routines.ClassificationRoutine)

```python
from torch_uncertainty.datamodules.classification.tiny_imagenet import TinyImageNetDataModule
from torchmetrics import Accuracy

acc = Accuracy("multiclass", num_classes=200)
data_module = TinyImageNetDataModule(
    root="data",
    batch_size=32,
)
model = load_model(0)

model.eval()
data_module.setup("test")

for batch in data_module.test_dataloader()[0]:
    x, y = batch
    y_hat = model(x)
    acc.update(y_hat, y)
print(f"Accuracy on the test set: {acc.compute():.3%}")
```

Put the model and the data on the GPU to make the computation faster.

If everything works, close this issue, and I'll add this function to the scripts folder of the HF dataset.

**Comment by nkotelevskii** [comment]
Dear Olivier,

Yes, it works, thank you very much!

The thing that confused me is that I used style="imagenet" for TinyImagenet, but not style="cifar".

Wish you all the best, and thank you again for sharing the weights!

**Comment by o-laurent** [comment]
You're very welcome!

All the best to you!
